### ISSUE TITLE
handoff + shunt: s1.mjs — shared System One runtime, rules backend first

### ISSUE THREAD

**Issue by MathBorgess** [MathBorgess/skills-catalog#32]
Wave 0 enabler. **Every other System One item depends on this one.** Ships with no model at all.

Parent: #31

## What to build

`skills/handoff/scripts/s1.mjs` and `skills/shunt/scripts/s1.mjs` — identical copies, exactly the arrangement `rtk.mjs` already uses, with `handoff.test.mjs` failing if they drift.

One interface, three call shapes, a swappable backend:

```js
choice(context, options, { backend })   // -> { index, label, p, dist }
score(context, levels,  { backend })    // -> { level, p, dist }
noul(context, question, { backend })    // -> { yes, p }
```

The backend contract is one method. This is lifted from `cua_s1`'s `PlanningBackend` protocol, and it is the reason the whole roadmap can ship before any model exists: the planner never knows what is behind it.

## Three backends, in this order

1. **`rules`** — reproduces today's behaviour *exactly*: the `GUARDED` regex, `LINE_MAX`/`BYTE_MAX`, `EFFORT_BY_TIER`. Returns `p` as a flat confidence (1.0 for a hard rule hit). **Zero behaviour change is the acceptance criterion for this issue.**
2. **`teacher`** — one cheap frontier call returning a typed choice. Lands with the item that first needs it.
3. **`local`** — safetensors + pure-JS forward pass. Lands with the trainer.

## Why pure JS for the `local` backend

`CLAUDE.md`: *"Scripts are Node ESM, standard library only."* The hook fires on every `Read|Edit|Write|NotebookEdit|Bash`, so a Python process or a native ONNX binary is out on both the rule and the latency.

At 706K parameters this is genuinely small: safetensors is a little-endian u64 header length, a JSON header, then raw tensor bytes — a `DataView` and typed-array views, no parser. The forward pass is a byte embedding lookup, two encoder layers, one option encoder, and an attention head — roughly five matmuls over 128-dim vectors.

The real cost is **proving equivalence**, and it must not be skipped: `norm_first=True`, the default activation of `TransformerEncoderLayer`, and LayerNorm ordering are all easy to get subtly wrong. Ship a fixture of reference logits and assert bit-for-bit, the same way the `rtk.mjs` drift test works.

## Every decision is logged

Each call appends one record through the shadow log (#33). Logging is not optional and not a debug flag — **it is the deliverable**. The rules backend exists to produce the dataset the later backends need.

## Done when

- [ ] `s1.mjs` exports `choice`, `score`, `noul` and a `rules` backend
- [ ] drift test fails when the two copies diverge
- [ ] a caller can swap backends without touching the call site
- [ ] no observable behaviour change anywhere in `handoff` or `shunt`
- [ ] every call emits one shadow-log record

## Depends on

Nothing. Blocks #33 and all of wave 1–4.
